src/Security/SecurityConfigUpdater.php

```diff
@@ -55,24 +75,47 @@ public function updateForAuthenticator(string $yamlSource, string $firewallName,
             $newData['security']['firewalls'] = [];
         }
 
-        if (!isset($newData['security']['firewalls'][$firewallName])) {
+        if (!isset($newData['security']['firewalls'][$firewallName]) && !$useSecurity51) {
             $newData['security']['firewalls'][$firewallName] = ['anonymous' => true];
+        } elseif (!isset($newData['security']['firewalls'][$firewallName]) && $useSecurity51) {
+            $newData['security']['firewalls'][$firewallName] = ['lazy' => true];
         }
 
         $firewall = $newData['security']['firewalls'][$firewallName];
 
-        if (!isset($firewall['guard'])) {
-            $firewall['guard'] = [];
-        }
+        if ($useSecurity51) {
+            if (isset($firewall['custom_authenticator'])) {
+                if (\is_array($firewall['custom_authenticator'])) {
+                    $firewall['custom_authenticator'][] = $authenticatorClass;
+                } else {
+                    $stringValue = $firewall['custom_authenticator'];
+                    $firewall['custom_authenticator'] = [];
+                    $firewall['custom_authenticator'][] = $stringValue;
+                    $firewall['custom_authenticator'][] = $authenticatorClass;
+                }
+            }
 
-        if (!isset($firewall['guard']['authenticators'])) {
-            $firewall['guard']['authenticators'] = [];
-        }
+            if (!isset($firewall['custom_authenticator'])) {
+                $firewall['custom_authenticator'] = $authenticatorClass;
+            }
+
+            if (!isset($firewall['entry_point'])) {
+                $firewall['entry_point'] = $authenticatorClass;
+            }
```

Comment:
Oh, hmm. This is not something I had thought of. I think it's not quite this simple. Hmm. Some info:

1) Not all custom authenticators provide an entrypoint. You need to implement an extra interface. Right now, our "login form" DOES, but our "blank" does not. Should that possibly be a question we ask for blank authenticators? Do you want to provide an entry point (but of course, we would need to explain what an entry point is)?

2) If the authenticator that we're generating DOES provide an entrypoint, then setting `entry_point` if it is not already set is a nice, safe thing to do. If the user did not previously have any "authentication mechanisms" (which includes things like http_basic and form_login) that provided an entry point, then setting this is technically unnecessary. But again, it seems like a safe thing to do.

3) However, if `entry_point` was NOT set... and you did already have exactly 1 previous "authentication mechanism" that provided an entry point, then this is a unique situation. By setting this, we are *changing* their entry point from the previous one to the new one. Probably... we should do this... because (for starters), we can't even know if they had previously a single auth mechanism that provided an entry point reliably... without doing some serious craziness. Anyways... this is a long way of saying... in this situation, we should STILL do exactly what you're doing here. But having a comment above `entry_point` in the config to explain things might help:

```yml
# the entry_point start() method determines what happens when an anonymous user access a protected page
```

Reply:
2 & 3 done

Reply:
done